discovery/service.go

```diff
@@ -430,19 +458,41 @@ func (d *Discovery) processNetworkAnnouncement(aMsg *networkMsg) bool {
 		}
 
 		if err := d.cfg.Router.AddNode(node); err != nil {
-			log.Errorf("unable to add node: %v", err)
-			return false
+			e := errors.Errorf("unable to add node: %v", err)
+			if routing.IsError(err,
+				routing.ErrOutdated,
+				routing.ErrIgnored) {
+				log.Info(e)
+			} else {
+				log.Error(e)
+			}
+			nMsg.err <- e
+			return nil
 		}
 
+		// Node announcement was successfully proceeded and know it
+		// might be broadcasted to other connected nodes.
+		announcements = append(announcements, msg)
+
+		nMsg.err <- nil
+		return announcements
+
 	// A new channel announcement has arrived, this indicates the
 	// *creation* of a new channel within the network. This only advertises
 	// the existence of a channel and not yet the routing policies in
 	// either direction of the channel.
 	case *lnwire.ChannelAnnouncement:
+		// TODO(andrew.shvv) remove after we get rid of the setting
+		// curve to nil log function
+		msg.NodeID1 = copyPubKey(msg.NodeID1)
```

Comment:
I moved the copy of public keys inside the validation function in order it will be more clear why we need to copy them.